blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubServerContainer.java

```diff
@@ -99,9 +99,10 @@
                 // once there is an easy way to delete existing GitHub servers this same logic should be added to validation above
                 String sanitizedUrl = discardQueryString(url);
                 Endpoint endpoint = new Endpoint(sanitizedUrl, name);
-                List<Endpoint> endpoints = Lists.newLinkedList(config.getEndpoints());
-                endpoints.add(endpoint);
-                config.setEndpoints(endpoints);
+                if (!config.addEndpoint(endpoint)) {
+                    message.add(new ErrorMessage.Error(GithubServer.API_URL, ErrorMessage.Error.ErrorCodes.ALREADY_EXISTS.toString(), GithubServer.API_URL + " is already registered as '" + endpoint.getName() + "'"));
```

Comment:
Might be better to move it up so that all errors are thrown together. e.g. if name is bad but the server url already exist.